packages/hash/api/src/model/comment.model.ts

```diff
@@ -0,0 +1,212 @@
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import {
+  Entity,
+  EntityConstructorArgs,
+  EntityExternalResolvers,
+  Comment,
+  UnresolvedGQLEntityType,
+  User,
+  Block,
+} from ".";
+import { DbClient } from "../db";
+import {
+  DbCommentProperties,
+  DbTextProperties,
+  EntityType,
+} from "../db/adapter";
+import {
+  SystemTypeName,
+  Comment as GQLComment,
+  LinkedEntityDefinition,
+} from "../graphql/apiTypes.gen";
+
+export type CommentExternalResolvers = EntityExternalResolvers | "contents"; // contents resolved in `src/graphql/resolvers/Comments/linkedEntities.ts`
+
+export type UnresolvedGQLComment = Omit<
+  GQLComment,
+  CommentExternalResolvers
+> & {
+  entityType: UnresolvedGQLEntityType;
+};
+
+export type CommentConstructorArgs = {
+  properties: DbCommentProperties;
+  outgoingEntityIds?: string[];
+} & EntityConstructorArgs;
+
+class __Comment extends Entity {
+  properties: DbCommentProperties;
+
+  constructor(args: CommentConstructorArgs) {
+    super(args);
+    this.properties = args.properties;
+  }
+
+  static async getEntityType(client: DbClient): Promise<EntityType> {
+    const commentEntityType = await client.getSystemTypeLatestVersion({
+      systemTypeName: "Comment",
+    });
+    return commentEntityType;
+  }
+
+  static async createComment(
+    client: DbClient,
+    params: {
+      createdBy: User;
+      parent: Block;
+      accountId: string;
+      content: TextToken[];
+    },
+  ): Promise<Comment> {
+    const { createdBy, parent, accountId, content } = params;
+
+    const textProperties: DbTextProperties = {
+      tokens: content,
+    };
+
+    const linkedEntities: LinkedEntityDefinition[] = [
+      {
+        path: "$.contents",
+        destinationAccountId: accountId,
+        entity: {
+          entityType: {
+            systemTypeName: SystemTypeName.Text,
+          },
+          entityProperties: textProperties,
+        },
+      },
+      {
+        path: "$.owner",
+        destinationAccountId: accountId,
+        entity: {
+          existingEntity: {
+            accountId: createdBy.accountId,
+            entityId: createdBy.entityId,
+          },
+        },
+      },
+      {
+        path: "$.parent",
+        destinationAccountId: accountId,
+        entity: {
+          existingEntity: {
+            accountId: parent.accountId,
+            entityId: parent.entityId,
+          },
+        },
+      },
+    ];
+
+    const commentProperties = {
+      createdAt: new Date().toISOString(),
+    };
+
+    const entity = await Entity.createEntityWithLinks(client, {
+      user: createdBy,
+      accountId,
+      entityDefinition: {
+        entityProperties: commentProperties,
+        versioned: true,
+        entityType: {
+          systemTypeName: SystemTypeName.Comment,
+        },
+        linkedEntities,
+      },
+    });
+
+    return new Comment({ ...entity, properties: commentProperties });
```

Comment:
Typescript doesn't know the type of  `entity.properties` because entity is not casted as a `Comment`, but as an `Entity`.
But this follows the same pattern that is used on `createPage` in `page.model` so I didn't give it much thought.
Do you have a suggestion on a better way to handle this?